src/main/java/use_cases/game_preparation_use_case/GamePrepInteractor.java

```diff
@@ -0,0 +1,199 @@
+package use_cases.game_preparation_use_case;
+
+import entities.GameEvent;
+import entities.GameState;
+import entities.Player;
+import entities.PlayerFactory;
+import entities.cardEffects.CardEffect;
+import entities.cardEffects.CardEffectFactory;
+import entities.cards.*;
+import entities.decks.DeckFactory;
+import entities.decks.EssenceDeck;
+import entities.decks.PlayerDeck;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class GamePrepInteractor implements GamePrepInputBoundary {
+
+    final GameState gameState;
+    final CardFactory cardFactory;
+    final DeckFactory deckFactory;
+    final PlayerFactory playerFactory;
+    final CardDataGateway dataAccessor;
+    final GamePrepOutputBoundary gamePrepPresenter;
+
+    public GamePrepInteractor(GameState gameState, CardFactory cardFactory,
+                              DeckFactory deckFactory, PlayerFactory playerFactory,
+                              CardDataGateway dataAccessor, GamePrepOutputBoundary gamePrepPresenter) {
+        this.gameState = gameState;
+        this.cardFactory = cardFactory;
+        this.deckFactory = deckFactory;
+        this.playerFactory = playerFactory;
+        this.dataAccessor = dataAccessor;
+        this.gamePrepPresenter = gamePrepPresenter;
+    }
+
+    /**
+     * Signal the presenter to display a prompt screen with all the available
+     * creature cards that the players can choose for the game, and ask the players
+     * to enter their names and select the creatures to play.
+     * This method will use the CardDataGateway to fetch all the game data, and
+     * it is the CardDataGateway's job to access the data stored outside the
+     * program.
+     */
+    @Override
+    public void promptPlayerInfo() {
+        gamePrepPresenter.showGamePreparationScreen(dataAccessor.getExistingCreatures());
+    }
+
+    /**
+     * Create the Player instances and the Creature instances based on the names
+     * provided in the {@code requestModel}, then signal the presenter to display
+     * the game screen.
+     * This method will send a GamePrepResponseModel containing all the values
+     * that need to be shown on the game screen.
+     *
+     * @param requestModel the GamePrepRequestModel containing the two players'
+     *                     names and their corresponding lists of creature names.
+     * @return a GamePrepResponseModel containing all necessary stats to be shown
+     * on screen (i.e. creature attack and hit-points, player names, etc.)
+     */
+    @Override
+    public GamePrepResponseModel processPlayerInfo(GamePrepRequestModel requestModel) {
+        if (requestModel.getPlayerName1().isEmpty() || requestModel.getPlayerName2().isEmpty()) {
```

Comment:
I believe we should also have more conditions to check for other types of usernames such as ones that are too long as it could effect UI/UX. Perhaps this could come in the form of an if statement checking for length
```suggestion
                 if(requestModel.getPlayerName1().helper() >10 || requestModel.getPlayerName2().helper() >10 ){
                              return gamePrepPresenter.displayErrorMessage("Please enter a shorter name for the player.");  
                  }
```
where helper() is a helper function to check length of player name and if player name is String then just .length()

Reply:
good point, and considering it's a small change I'll make a note of it for now but add this change later, I know the other use cases also have minor changes so we'll just create a separate branch. But that'll be after we have these fundamental features all merged.